test/collector.test.ts

```diff
@@ -11,33 +11,28 @@ import { Wallet } from '@ethersproject/wallet';
 
 use(solidity);
 
-const FIRST_PARTNER_SHARE = 20;
-const SECOND_PARTNER_SHARE = 30;
-const THIRD_PARTNER_SHARE = 40;
-const FOURTH_PARTNER_SHARE = 10;
-
-const buildPartners = (partners: Wallet[], shares?: number[]) => {
-    return [
-        {
-            beneficiary: partners[0].getAddress(),
-            share: shares?.[0] ?? FIRST_PARTNER_SHARE
-        },
-        {
-            beneficiary: partners[1].getAddress(),
-            share: shares?.[1] ?? SECOND_PARTNER_SHARE
-        },
-        {
-            beneficiary: partners[2].getAddress(),
-            share: shares?.[2] ?? THIRD_PARTNER_SHARE
-        },
-        {
-            beneficiary: partners[3].getAddress(),
-            share: shares?.[3] ?? FOURTH_PARTNER_SHARE
-        }
-    ];
+const PARTNER_SHARES = [20, 30, 40, 10];
+const NUMBER_OF_PARTNERS = 4;
+
+type Partner = {
+    beneficiary: string;
+    share: number;
 };
 
-async function prepareAll(wallets: Wallet[]) {
+async function buildPartners(parternWallets: Wallet[], shares?: number[]) {
+    const partners: Partner[] = [];
+
+    for (let i = 0; i < NUMBER_OF_PARTNERS; i++) {
+        partners.push({
+            beneficiary: await parternWallets[i].getAddress(),
+            share: shares?.[i] ?? PARTNER_SHARES[i]
+        });
+    }
```

Comment:
please please prefer map to for loops. Main reasons are:
1. immutability
2. map, filter, etc are optimised, for loop is not

The first reason is obviously the killer of for loops unless you really really REALLY need the performance increase (for small arrays it's pretty much negligible)

Reply:
Done.